src/RamCondition.h

```diff
@@ -336,130 +273,113 @@ class RamExistenceCheck : public RamCondition {
-        // arguments.size() - 1 because we discard the height annotation
-        for (unsigned i = 0; i < arguments.size() - 1; i++) {
-            if (arguments[i]) {
-                res |= (1 << i);
+    /** Create clone */
+    RamExistenceCheck* clone() const override {
+        std::vector<std::unique_ptr<RamValue>> newValues;
+        for (auto& cur : values) {
+            RamValue* val = nullptr;
+            if (cur != nullptr) {
+                val = cur->clone();
             }
+            newValues.push_back(std::unique_ptr<RamValue>(val));
         }
+        RamExistenceCheck* res = new RamExistenceCheck(
+                std::unique_ptr<RamRelationReference>(relation->clone()), std::move(newValues));
         return res;
     }
 
-    /** Is key total */
-    bool isTotal() const {
-        for (const auto& cur : arguments) {
-            if (!cur) {
-                return false;
-            }
-        }
-        return true;
+protected:
+    /** Check equality */
+    bool equal(const RamNode& node) const override {
+        assert(nullptr != dynamic_cast<const RamExistenceCheck*>(&node));
+        return RamAbstractExistenceCheck::equal(node);
     }
+};
 
-    /** Obtain list of child nodes */
-    std::vector<const RamNode*> getChildNodes() const override {
-        std::vector<const RamNode*> res = {relation.get()};
-        for (const auto& cur : arguments) {
-            res.push_back(cur.get());
-        }
-        return res;
+/**
+ * Existence check for a relation for provenance existence check
+ */
+class RamProvenanceExistenceCheck : public RamAbstractExistenceCheck {
+public:
+    RamProvenanceExistenceCheck(
+            std::unique_ptr<RamRelationReference> rel, std::vector<std::unique_ptr<RamValue>> vals)
+            : RamAbstractExistenceCheck(RN_ProvenanceExistenceCheck, std::move(rel), std::move(vals)) {}
+
+    /** Print */
+    void print(std::ostream& os) const override {
+        os << "("
+           << join(values, ",",
+                      [](std::ostream& out, const std::unique_ptr<RamValue>& value) {
+                          if (!value) {
+                              out << "_";
+                          } else {
+                              out << *value;
+                          }
+                      })
+           << ") prov∈ " << relation->getName();
     }
 
     /** Create clone */
     RamProvenanceExistenceCheck* clone() const override {
-        RamProvenanceExistenceCheck* res =
-                new RamProvenanceExistenceCheck(std::unique_ptr<RamRelationReference>(relation->clone()));
-        for (auto& cur : arguments) {
+        std::vector<std::unique_ptr<RamValue>> newValues;
+        for (auto& cur : values) {
             RamValue* val = nullptr;
             if (cur != nullptr) {
                 val = cur->clone();
             }
-            res->arguments.emplace_back(val);
+            newValues.push_back(std::unique_ptr<RamValue>(val));
```

Comment:
Since you're putting a RamValue into a vector of unique pointers to RamValues, you can just do
```suggestion
            newValues.emplace_back(val);
```